core/src/test/kotlin/fr/sncf/osrd/stdcm/VisitedNodesTests.kt

```diff
@@ -79,17 +128,120 @@ class VisitedNodesTests {
         val params1 =
             VisitedNodes.Parameters(
                 fingerprint = fingerprint,
-                startTime = 0.0,
-                duration = 100.0,
+                timeData =
+                    TimeData(
+                        earliestReachableTime = 0.0,
+                        maxDepartureDelayingWithoutConflict = 100.0,
+                        timeOfNextConflictAtLocation = 100.0,
+                        totalRunningTime = 0.0,
+                        departureTime = 0.0,
+                        stopTimeData = listOf(),
+                    ),
                 maxMarginDuration = 100.0,
-                baseNodeCost = 0.0,
             )
         // At t=120, it's covered by the first parameters at cost=20
-        val params2 = params1.copy(startTime = 120.0, baseNodeCost = 15.0)
+        val params2 =
+            params1.copy(
+                timeData =
+                    params1.timeData.copy(
+                        earliestReachableTime = 120.0,
+                        totalRunningTime = 15.0,
+                    )
+            )
         // The duration here is longer than the previous test: it's visited from t=120 to t=125
         // (above the cost function), then unvisited from t=125. Some parts are unvisited => false
         // is expected
         visitedNodes.markAsVisited(params1)
         assertFalse { visitedNodes.isVisited(params2) }
     }
+
+    @ParameterizedTest
+    @ValueSource(booleans = [true, false])
+    fun testStopTimes(lowerStopDuration: Boolean) {
```

Comment:
I feel like we're missing some test cases ? I don't think we're testing out the visited with margins case ?

Reply:
Fixed in 0c57be6390f29f51b60b1078138c60211d4dfd1f

It did find some weird stuff, not related to the margin specifically but I hadn't try with 0-length visiting ranges